src/server.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+
+#
+# server.py
+# Created by Stephen Brimhall on 1/23/17
+# Copyright (c) SST CTF and Stephen Brimhall. All Rights Reserved
+#
+# This file implements the `server` class for the chatd instant messaging system
+
+# Begin system imports
+import sys;
+import os;
+import socket;
+import threading;
+# End system imports
+
+# Begin local imports
+from src.connection import Connection as Client;
+from src.queue import Queue as MessageQueue;
+# End local imports
+
+# Begin class
+class ChatdServer(object):
+    # ChatdServer -- Manage chat threads and socket objects to facilitate
+    # communication between all clients
+
+    def __init__(self):
+        # Takes no arguments, returns new ChatdServer object
+
+        # Initialize server socket
+        self.serverSocket = socket.socket();
+
+        # Initialize client list
+        self.clients = [];
+
+        # Initialize client list lock
+        self.clientLock = threading.RLock();
+
+        # Initialize global message queue
+        self.globalMessageQueue = MessageQueue();
+
+    def bind(self, port):
+        # Binds server socket to all interfaces on given port. Expects UInt16.
+
+        # Bind server to all interfaces
+        self.serverSocket.bind(("0.0.0.0",port));
```

Comment:
How would the user know the crash was caused by a port config problem instead os something else. At least give the user a print out error!